packages/react-compose/src/compose.ts

```diff
@@ -20,11 +20,11 @@ function compose<
     wasComposedPreviously(input) ? input.fluentComposeConfig : undefined,
   );
 
-  const Component = (React.forwardRef<TElementType, TInputProps & TParentProps & { as?: React.ElementType }>(
+  const Component = (React.forwardRef<HTMLElement, TInputProps & TParentProps & { as?: React.ElementType }>(
```

Comment:
no, the previous type is wrong. `TElementType` can be something like `div` but in that case what we actually want here is `React.forwardRef<HTMLDivElement` instead of `React.forwardRef<TElementType`. 
Honestly, typings here is quite messed up in general (look at how many `as` being used around this code). I don't want to change the world here so taking a smaller step to correct this